src/emqtt_bench.erl

```diff
@@ -1231,22 +1270,71 @@ counters() ->
    Idxs = lists:seq(2, length(Names) + 1),
    lists:zip(Names, Idxs).
 
+maybe_init_prometheus(false) ->
+    ok;
+maybe_init_prometheus(true) ->
+    Collectors = [ prometheus_boolean,
+                   prometheus_counter,
+                   prometheus_gauge,
+                   prometheus_histogram,
+                   prometheus_quantile_summary,
+                   prometheus_summary],
+    application:set_env(prometheus, collectors, Collectors),
+    {ok, _} = application:ensure_all_started(prometheus),
+    Counters = [ publish_latency
+               , recv
+               , sub
+               , sub_fail
+               , pub
+               , pub_fail
+               , pub_overrun
+               , pub_succ
+               , connect_succ
+               , connect_fail
+               , connect_retried
+               , reconnect_succ
+               , unreachable
+               , connection_refused
+               , connection_timeout
+               , connection_idle
+               ],
```

Comment:
Nit: maybe we could move this list of counter names to a `counter_name/0` function, and use it both here and in `counters/0`?

Reply:
Will do in the next PR. There are a few more things to deduplicate.